doc/users.md

```diff
@@ -1 +1,1712 @@
+- 5 = ALL-WMS
+- 6 = WMS-NETWORK DEV
+- 7 = WMS-SE
+- 8 = WMS-Salesman PMD
+
+for edit, you need get :
+- if type_akses given then company_id, branch_id, region_id, kios_id is required depend on the value.
+- company_id from companies tables if type_akses in range 1-4.
+- branch_id from branches tables if type_akses in range 3-4.
+- region_id from regions tables if type_akses is 2.
+- kios_id from kios tables if type_akses is 4.
+
+#### Validation
+- name: not empty
+- type_akses: must in range (1-8)
+- company_id: required and not empty and must be registered in companies if type_akses in range (1-4)
+- region_id: required and not empty and must be registered in regions if type_akses = 2
+- branch_id: required and not empty and must be registered in branches if type_akses in range (3-4)
+- kios_id: required and not empty and must be registered in kios if type_akses = 4
+
+### Scenario Test
+
+#### Case : Negative Case 1
+
+Request Payload : empty
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "cdc-200",
+    "status_message": "no change is requested",
+    "data": null
+}
+```
+
+#### Case : Negative Case 2
+
+Request Payload :
+```
+{}
+```
+
+Response HTTP Status Code : 200
+
+Response Payload :
+```
+{
+    "status_code": "cdc-200",
+    "status_message": "no change is requested",
+    "data": null
+}
+```
+
+#### Case : Negative Case 3
+
+Request payload :
+```
+{
+    "name": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload :
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "name is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 4
+
+Request Payload
+```
+{
+    "type_akses": 9
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "type_akses must in range 1-8",
+    "data": null
+}
+```
+
+#### Case : Negative Case 5
+
+Request Payload
+```
+{
+    "type_akses": 1
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "company_id is required",
+    "data": null
+}
+```
+
+#### Case : Negative Case 6
+
+Request Payload
+```
+{
+    "type_akses": 1,
+    "company_id": ""
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "company_id is empty",
+    "data": null
+}
+```
+
+#### Case : Negative Case 7
+
+Request Payload
+```
+{
+    "type_akses": 1,
+    "company_id": "test"
+}
+```
+
+Response HTTP Status Code : 404
+
+Response Payload
+```
+{
+    "status_code": "cdc-404",
+    "status_message": "company_id not found in companies",
+    "data": null
+}
+```
+
+#### Case : Negative Case 8
+
+Request Payload
+```
+{
+    "type_akses": 2,
+    "company_id": "5fb5df66-5f80-416d-a055-401bcf944ef7"
+}
+```
+
+Response HTTP Status Code : 400
+
+Response Payload
+```
+{
+    "status_code": "cdc-400",
+    "status_message": "region_id is required",
```

Comment:
in edit mode, if there is no region_id, you can use the existing data. it's mean the region_id is not changed. But if any update of company_id, you should revalidate : "whether the existing region_id is still belongs to company_id"?